lib/engine/game/g_1849/step/buy_token.rb

```diff
@@ -0,0 +1,154 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/base'
+
+module Engine
+  module Game
+    module G1849
+      module Step
+        class BuyToken < Engine::Step::Base
+          ACTIONS = %w[buy_token pass].freeze
+          MIN_PRICE = 1
+
+          def actions(entity)
+            return [] unless entity == current_entity
+            return [] unless can_buy_token?(entity)
+
+            ACTIONS
+          end
+
+          def round_state
+            super.merge(
+              {
+                bought_token: false,
```

Comment:
You don't need this in round state. It is only used within this step. Just do something like `@bought_token = false` in `def setup`.

Reply:
Looks like there is some confusion. I'm saying that `bought_token` should not be added to round at all. It is only used here, so it doesn't need to be available outside of this step.

Reply:
got it. Fixed now.